qsimcirq/qsim_simulator.py

```diff
@@ -376,27 +376,34 @@ def _sample_measure_results(
                 seed=self._prng,
             )
 
-            for i in range(repetitions):
-                for key, op in meas_ops.items():
-                    meas_indices = [qubit_map[qubit] for qubit in op.qubits]
-                    invert_mask = op.gate.full_invert_mask()
-                    for j, q in enumerate(meas_indices):
-                        results[key][i][j] = full_results[i][q] ^ invert_mask[j]
+            for key, op in meas_ops.items():
+                meas_indices = [qubit_map[qubit] for qubit in op.qubits]
+                invert_mask = op.gate.full_invert_mask()
+                # Match result order to order in ops, then apply invert mask
+                permuted_results = full_results[:, meas_indices]
+                results[key] = np.logical_xor(permuted_results, invert_mask)
```

Comment:
I think the `^` operator should work fine on arrays:
```suggestion
                results[key] = permuted_results ^ invert_mask
```

Reply:
Got `TypeError: ufunc 'bitwise_xor' not supported for the input types` - maybe because `invert_mask` isn't an `np.array`? Sticking with `np.logical_xor` for now.

Reply:
Hm, we use `^` internally, so I'm wondering what the difference is here. What is the dtype of the `measurements` array? Also, it looks like `np.logical_xor` coerces the output to be a bool array, while `^` (which calls `np.bitwise_xor` under the hood) does not:
```
In [11]: np.zeros((3, 2), dtype=np.uint8) ^ [True, False]
Out[11]: 
array([[1, 0],
       [1, 0],
       [1, 0]], dtype=uint8)

In [12]: np.logical_xor(np.zeros((3, 2), dtype=np.uint8), [True, False])
Out[12]: 
array([[ True, False],
       [ True, False],
       [ True, False]])
```

Reply:
Good call on the `dtype` - it was defaulting to `float64` which didn't want to XOR with booleans.